driver-core/src/main/java/com/datastax/driver/core/ColumnDefinitions.java

```diff
@@ -307,6 +308,23 @@ void checkType(int i, DataType.Name name) {
         return defined.getName();
     }
 
+    CodecRegistry getCodecRegistry() {
+        if(codecRegistry == null)
```

Comment:
I think that ColumnDefinitions is not meant to be shared by multiple threads. By the time `setCodecRegistry()` is called on it (i.e. from `DefaultPreparedStatement.fromMessage` and `ArrayBackedResultSet.fromMessage`) no other thread can access the instance; however, we can definitely make the field volatile to be on the safe side.